src/main/java/edu/harvard/iq/dataverse/engine/command/impl/PublishDatasetCommand.java

```diff
@@ -218,9 +222,23 @@ private void verifyCommandArguments(CommandContext ctxt) throws IllegalCommandEx
             if (minorRelease && !getDataset().getLatestVersion().isMinorUpdate()) {
                 throw new IllegalCommandException("Cannot release as minor version. Re-try as major release.", this);
             }
+
+            if (getDataset().getFiles().isEmpty() && requiresFilesToPublishDataset()) {
+                throw new IllegalCommandException(BundleUtil.getStringFromBundle("dataset.mayNotPublish.FilesRequired"), this);
+            }
         }
     }
-    
+    private boolean requiresFilesToPublishDataset() {
```

Comment:
As noted in the docs, this means that if the root is set to requiring files you can't turn it off anywhere. Would it be possible to make this setting like many of our others where the choice for any given collection is on/off/inherit? There should be lots of code examples.

Reply:
I think I've seen examples where null=inherit and true/false are specific to that Dataverse. Good idea. I'll implement that.